compiler/ballerina-parser/src/test/java/io/ballerinalang/compiler/parser/test/syntax/expressions/CheckExpressionTest.java

```diff
@@ -47,4 +67,34 @@ public void testCheckWithMissingExpr() {
     public void testCheckWithMissingBinaryOp() {
         testFile("check-expr/check_expr_source_04.bal", "check-expr/check_expr_assert_04.json");
     }
+
+    @Test
+    public void testOnFailCheckWithMissingFail() {
+        testFile("on-fail-check/on_fail_check_source_02.bal", "on-fail-check/on_fail_check_assert_02.json");
+    }
+
+    @Test
+    public void testOnFailCheckpanicWithoutDoubleArrow() {
+        testFile("on-fail-check/on_fail_check_source_03.bal", "on-fail-check/on_fail_check_assert_03.json");
+    }
+
+    @Test
+    public void testOnFailWithSelect() {
+        testFile("on-fail-check/on_fail_check_source_07.bal", "on-fail-check/on_fail_check_assert_07.json");
+    }
+
+    @Test
+    public void testOnFailWithOnConflict() {
+        testFile("on-fail-check/on_fail_check_source_08.bal", "on-fail-check/on_fail_check_assert_08.json");
+    }
+
+    @Test
+    public void testOnFailWithJoin() {
+        testFile("on-fail-check/on_fail_check_source_09.bal", "on-fail-check/on_fail_check_assert_09.json");
+    }
+
+    @Test
+    public void testOnFailCheckPanicWithEquals() {
+        testFile("on-fail-check/on_fail_check_source_10.bal", "on-fail-check/on_fail_check_assert_10.json");
+    }
 }
```

Comment:
Let's also add a test with extra tokens to check recovery.
e.g. `    _ = check x on fail % e & => /error("Error!");`